Sources/ClientRuntime/Networking/Http/UnknownHttpServiceError.swift

```diff
@@ -4,40 +4,39 @@
  */
 
 /// General Service Error structure used when exact error could not be deduced from the `HttpResponse`
-public struct UnknownHttpServiceError: HttpServiceError, Swift.Equatable {
-    public var _errorType: String?
+public struct UnknownHTTPServiceError: ServiceError, HTTPError, Error {
+    public var typeName: String?
 
-    public var _isThrottling: Bool = false
+    public var message: String?
 
-    public var _statusCode: HttpStatusCode?
-
-    public var _headers: Headers?
-
-    public var _message: String?
-
-    public var _retryable: Bool = false
-
-    public var _type: ErrorType = .unknown
+    public var httpResponse: HttpResponse
 }
 
-extension UnknownHttpServiceError {
+extension UnknownHTTPServiceError {
 
-    /// Creates an `UnknownHttpServiceError` from a HTTP response.
+    /// Creates an `UnknownHTTPServiceError` from a HTTP response.
     /// - Parameters:
     ///   - httpResponse: The `HttpResponse` for this error.
     ///   - message: The message associated with this error. Defaults to `nil`.
     ///   - errorType: The error type associated with this error.  Defaults to `nil`.
-    public init(httpResponse: HttpResponse, message: String? = nil, errorType: String? = nil) {
-        self._statusCode = httpResponse.statusCode
-        self._headers = httpResponse.headers
-        self._message = message
-        self._errorType = errorType
+    public init(httpResponse: HttpResponse, message: String? = nil, typeName: String? = nil) {
+        self.typeName = typeName
+        self.httpResponse = httpResponse
+        self.message = message
     }
 }
 
-extension UnknownHttpServiceError: WaiterTypedError {
-
-    /// The Smithy identifier, without namespace, for the type of this error, or `nil` if the
-    /// error has no known type.
-    public var waiterErrorType: String? { _errorType }
+extension UnknownHTTPServiceError {
+
+    public static func makeError(
```

Comment:
This is the function that is called when the service returns an error but the type of the error is not one of the predefined / modeled error types.

For now this always returns `UnknownHTTPServiceError` but in the future I may return other types for specific failures.